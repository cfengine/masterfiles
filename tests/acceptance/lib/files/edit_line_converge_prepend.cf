body common control
{
      inputs => { '../../default.cf.sub' };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}
#######################################################

bundle common classes
{
  classes:

      "testing_masterfiles_policy_framework"
        scope => "namespace",
        comment => "This class is needed so that dcs.cf.sub includes the stdlib instead of using plucked.cf.sub from core which might get out of date and cause us to not test current code from the MPF.";
}

bundle agent init
{
  files:
      # The tested file "actual" is copied from our seeded starting position.
      "$(G.testfile)"
      copy_from => local_cp("$(this.promise_filename).start");

}

#######################################################

bundle agent test
{
  meta:
      "description" -> { "CFE-3483" }
        string => "Test that converge_prepend works as expected";

  files:
      "$(G.testfile)"
        edit_line => converge_prepend( "covfefe", "Despite the constant negative press covfefe" );

}

#######################################################

bundle agent check
{
  methods:
      "Pass/FAIL"
        usebundle => dcs_check_diff($(G.testfile), "$(this.promise_filename).expected", $(this.promise_filename));
}
