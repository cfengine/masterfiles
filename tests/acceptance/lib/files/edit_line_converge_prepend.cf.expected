Despite the constant negative press covfefe
It has nothing to do with the oranges of the investigation, global waming, or hamberders
